File: java/rotational-cipher/src/main/java/RotationalCipher.java
import java.util.stream.Collectors;

class RotationalCipher {
    Integer offset;

    RotationalCipher(int shiftKey) {
        offset = shiftKey;
    }

    String rotate(String data) {
        return data.chars().map(this::rot).mapToObj(c -> String.valueOf((char) c)).collect(Collectors.joining(""));
    }

    private int rot(int c) {
        if (Character.isLetter(c)) {
            return c >= 'a' ? (c - 'a' + offset) % 26 + 'a' : (c - 'A' + offset) % 26 + 'A';
        }
        return c;
    }
}

File: go/rotational-cipher/rotational_cipher.go
package rotationalcipher

func RotationalCipher(s string, rot int) string {
	var r []rune
	for _, c := range s {
		base := 0
		switch {
		case c >= 'A' && c <= 'Z':
			base = int('A')
		case c >= 'a' && c <= 'z':
			base = int('a')
		}
		if base != 0 {
			c = rune((int(c) - base + rot) % 26 + base)
		}
		r = append(r, c)
	}
	return string(r)
}
